### PR TITLE
PLT-6408 Framework for job server

### PR DESCRIPTION
**Pull request by hmhealey**
Here's the framework for the job server that will be able to either run jobs (data retention, search indexing, etc) as either part of the main Mattermost server or from a standalone executable. I had to make some changes to the enterprise imports for this:
1. Instead of being included by cmd/platform/imports.go, there's now a separate imports package so they can be included when there's a different entry point into the program.
2. As I mentioned a couple weeks ago, we had to put the enterprise interface for jobs in the `einterfaces/jobs` package to avoid an import cycle of `einterfaces > store > einterfaces`

And a few things to note about the job server itself:
1. The job server can be built using `make build-job-server` and ran with `make run-job-server`
2. The code to run the job server is in the `jobs` package and is ran by both the platform and standalone job server located in the `jobs/jobserver` package.
3. The dependencies are set up in such a way that the `jobs` package should only rely on the `model`, `utils`, and `store` packages however it could probably also use `app` code should we need it.
4. The job server (standalone or otherwise) listens for config changes through a new listener system so that the utils package doesn't need to explicitly manage changes to the jobs like it does with the logger, SAML, LDAP, etc.
5. The code around setting the jobs status and such will be implemented by the individual jobs, so it is currently unused.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-6408

#### Checklist
- Added or updated unit tests
- Added API documentation (https://github.com/mattermost/mattermost-api-reference/pull/238)
- All new/modified APIs include changes to the drivers
- Has enterprise changes (PLT-6408 Framework for job server)
- Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates
- Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

**Comment by grundleborg** [comment]
I've done a cursory review of the general architecture of this, and it looks good to me. Will try and take a more detailed look at it later.

File: api4/context.go [hunk 1]
```diff
@@ -536,3 +536,27 @@ func (c *Context) RequireCommandId() *Context {
 	}
 	return c
 }
+
+func (c *Context) RequireJobId() *Context {
+	if c.Err != nil {
+		return c
+	}
+
+	l4g.Error("job_id=" + c.Params.JobId)
```
Comment (jwilander) [inline api4/context.go]: Debug statement you forgot to remove?

File: api4/context.go [hunk 2]
```diff
@@ -536,3 +536,27 @@ func (c *Context) RequireCommandId() *Context {
...
+	}
+	return c
+}
+
+func (c *Context) RequireJobType() *Context {
+	if c.Err != nil {
+		return c
+	}
+
+	l4g.Error("job_type=" + c.Params.JobType)
```
Comment (jwilander) [inline api4/context.go]: Here too

File: api4/job.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import (
+	"net/http"
+
+	l4g "github.com/alecthomas/log4go"
+	"github.com/mattermost/platform/app"
+	"github.com/mattermost/platform/model"
+)
+
+func InitJob() {
+	l4g.Error("Initializing post API routes")
```
Comment (jwilander) [inline api4/job.go]: Shouldn't this be `Info` and plus copy/paste error saying `post`

**Pull request merged**